### PR TITLE
Fix WASM build

### LINKED ISSUE #276: WASM crashes
So I made sure I was on wgpu 15 and using the most recent version.

I cannot seem to make the run_wasm work. On the vello example, I see a problem with adapter.features() from webgpu causing a crash. Doesn't work on canary or nightly.

image

In my personal project I get a bug saying there's no way to allocate a storage buffer. Unrelated, but just letting you know. Perhaps this could be a 2 birds with 1 stone issue.

image

### PR DESCRIPTION
**Pull request by armansito**
- Rolled wgpu to 0.16.
- Incorprated the [instant crate](https://crates.io/crates/instant) in lieu of `std::time::Instant` which works on WASM and native builds.
- Fixed the issue with window scaling by setting the canvas size based on winit Window dimensions.

This resolves #276

### CODE REVIEW

File: examples/scenes/Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ vello_svg = { path = "../../integrations/vello_svg" }
 anyhow = { workspace = true }
 clap = { workspace = true, features = ["derive"] }
 image = "0.24.5"
+instant = "0.1.12"
```
Comment (DJMcNab) [inline examples/scenes/Cargo.toml:19]: According to `instant`'s documentation, one of the `stdweb` or `wasm-bindgen` features need to be updated.

I still don't have a working browser with WebGPU, so maybe this works? Perhaps `instant`'s docs need updating

File: src/engine.rs [hunk 2]
```diff
@@ -311,8 +311,8 @@ impl Engine {
                         bytes,
                         wgpu::ImageDataLayout {
                             offset: 0,
-                            bytes_per_row: NonZeroU32::new(
-                                image_proxy.width * format.describe().block_size as u32,
+                            bytes_per_row: Some(
+                                image_proxy.width * format.block_size(None).unwrap(),
```
Comment (DJMcNab) [inline src/engine.rs:315]:
```suggestion
                                image_proxy.width * format.block_size(None).expect("All supported ImageFormats have a valid block size"),
```
or words to that effect

**Review by DJMcNab — COMMENTED** [review]
Code seems reasonable to me, although I can't test it without a WebGPU compatible browser.

File: examples/scenes/Cargo.toml (hunk 1 above)
Comment (armansito) [reply examples/scenes/Cargo.toml]: Good catch. I went ahead and enabled `wasm-bindgen` though AFAICT all the animated scenes still worked without it.

File: src/engine.rs (hunk 2 above)
Comment (armansito) [reply src/engine.rs]: Done.

**Comment by armansito** [comment]
Unfortunately this PR breaks the bevy example since bevy_render currently depends on wgpu 0.15.

**Review by dfrg — APPROVED** [review]

**Comment by armansito** [comment]
> All the changes here are good, there are just a few additions I think should be included in this PR.
> 
> I think this can probably incorporate the changes from #301. In particular, that's an update of `cargo-run-wasm`, including the new support for the `-p` shorthand; adding the WASM target to the instructions is optional but would be nice.
> 
> Also, I'd like us to add a Warning to the Bevy section of the README that the example temporarily doesn't compile. (For context, we decided on Zulip that the Bevy example shouldn't block needed changes, so long as there's a reasonable expectation it won't be broken for long)
> 
> > Warning
> > This example currently does not compile. We expect to resolve this as soon as bevy's has updated to wgpu 0.16
> 
> Additionally, we specify the wgpu version in a badge in README.md. This also needs to be updated. There should probably also be a comment to this effect in the root Cargo.toml.

Done. I cherry-picked your change from #301 into the PR and updated the WASM build instructions to say `cargo run_wasm -p with_winit --bin with_winit_bin` (as the `--bin` part is still necessary for it to work correcfly).

File: README.md [hunk 3]
```diff
@@ -65,6 +65,9 @@ cargo run -p with_winit -- download
 
 ### Bevy
 
+> **Warning**
```
Comment (DJMcNab) [inline README.md:68]:
```suggestion
> **Warning**  
```

**Review by DJMcNab — APPROVED** [review]
So long as the most recent changes have been tested (i.e. in a browser), this seems all good

**Pull request merged**